dsperse/src/run/utils/runner_utils.py

```
@@ -76,12 +76,97 @@ def prepare_slice_io(run_dir: Path, slice_id: str) -> tuple[Path, Path]:
+        if path.exists():
+            return str(path)
+
+        # Fallback to sibling/parent if it doesn't exist (legacy behavior in some tests/backends)
+        alt_path = (base_dir.parent / p_str).resolve()
+        if alt_path.exists():
+            return str(alt_path)
+
+        return str(path)
+
+    @staticmethod
+    def run_onnx_slice(slice_info: dict, input_tensor_path, output_tensor_path, slice_dir: Path = None):
```

Ruff (0.14.11)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)